src/simplejavacalculator/Calculator.java

```diff
@@ -105,22 +105,35 @@ public Double calculateMono(MonoOperatorModes newMode, Double num) {
             }
             if (num % 90 == 0 && num % 180 != 0)
                 return NaN;
-            }
+            
 
             return Math.tan(num);
         }
-        if (newMode == MonoOperatorModes.log) {
+
+        if (newMode == MonoOperatorModes.log){
             return log10(num);
         }
-        if (newMode == MonoOperatorModes.rate) {
+        if (newMode == MonoOperatorModes.rate){
            return num / 100;
         }
         if (newMode == MonoOperatorModes.abs){
             return Math.abs(num);
         }
+        if(newMode ==MonoOperatorModes.factorial)
+        {   
+            if(num==0){
+                num=1.0;
+                return num;
+            }
+            int factorial=(int)Math.round(num);
```

Comment:
Can you come with a more proper and elegant way to do this? What do you think? 
Finally, the method here is already way too long and not very readable. I would prefer refactoring into small pieces rather than having a new thing and loop in there.